### PR TITLE
[inner-1959] adjust 'xa prepare' code

### PR DESCRIPTION
**Pull request by wenyh1**
Reason:  
  BUG inner-1959
Type:  
  BUG/Improve  
Influences：  
   fix xx

### CODE REVIEW

File: src/main/java/com/actiontech/dble/backend/mysql/nio/handler/transaction/xa/handler/AbstractXAHandler.java [hunk 1]
```diff
@@ -99,58 +109,75 @@ protected boolean decrementToZero(RouteResultsetNode rrsn) {
...
-            }
-            this.setFail(closeReason);
-            currentStage.onConnectionClose((MySQLResponseService) service);
-            if (finished) {
-                changeStageTo(next());
-            }
-        } else {
-            if (logger.isDebugEnabled()) {
-                logger.debug("conn[backendId=" + service.getConnection().getId() + "] was closed in gap of two stage");
+        lock.lock();
```
Comment (PanternBao) [inline src/main/java/com/actiontech/dble/backend/mysql/nio/handler/transaction/xa/handler/AbstractXAHandler.java:134]: why add lock?

File: src/main/java/com/actiontech/dble/backend/mysql/nio/handler/transaction/xa/handler/AbstractXAHandler.java [hunk 2]
```diff
@@ -99,58 +109,75 @@ protected boolean decrementToZero(RouteResultsetNode rrsn) {
...
     public void connectionError(Throwable e, Object attachment) {
-        logger.warn("connection Error in xa transaction, err:", e);
-        boolean finished;
         lock.lock();
         try {
-            errorConnsCnt++;
-            finished = canResponse();
+            logger.warn("connection Error in xa transaction, err:", e);
+            boolean finished;
+            lock.lock();
```
Comment (PanternBao) [inline src/main/java/com/actiontech/dble/backend/mysql/nio/handler/transaction/xa/handler/AbstractXAHandler.java:166]: why add lock in lock

Comment (wenyh1) [reply src/main/java/com/actiontech/dble/backend/mysql/nio/handler/transaction/xa/handler/AbstractXAHandler.java:166]: this need adjsut

**Comment by wenyh1** [comment]
to draft reason：
Also need to consider other more multi-node response scenario(not only in xa).
Need to be further discussed, the final repair plan.

**Comment by wenyh1** [comment]
there is a new PR to replace

**Pull request closed without merging**